pkg/template/processor_test.go

```diff
@@ -0,0 +1,223 @@
+package template_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/codeready-toolchain/member-operator/pkg/template"
+
+	"github.com/go-logr/zapr"
+	apitemplate "github.com/openshift/api/template/v1"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"fmt"
+	testtemplates "github.com/codeready-toolchain/member-operator/test/templates"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"context"
+	authv1 "github.com/openshift/api/authorization/v1"
+	projectv1 "github.com/openshift/api/project/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"github.com/satori/go.uuid"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+)
+
+func TestProcess(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		// given
+		scheme := runtime.NewScheme()
+		utilruntime.Must(apitemplate.Install(scheme)) // see https://github.com/openshift/oc/blob/master/cmd/oc/oc.go#L77
+		values := map[string]string{
+			"PROJECT_NAME": "foo",
+			"COMMIT":       "1a2b3c",
+			"USER_NAME":    "developer",
+		}
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, scheme, logger(t))
+
+		// when
+		objs, err := p.Process(templateContent(t), values)
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, objs, 2)
+
+		// project request
+		projectKind := objs[0].Object.GetObjectKind()
+		require.IsType(t, &unstructured.Unstructured{}, projectKind)
+		projectRequest := projectKind.(*unstructured.Unstructured)
+		prJson, err := projectRequest.MarshalJSON()
+		require.NoError(t, err, "failed to marshal json for projectrequest")
+		assert.Equal(t, expectedProjectRequest(), string(prJson))
+
+		// role binding
+		rbKind := objs[1].Object.GetObjectKind()
+		require.IsType(t, &unstructured.Unstructured{}, rbKind)
+		roleBinding := rbKind.(*unstructured.Unstructured)
+		rbJson, err := roleBinding.MarshalJSON()
+		require.NoError(t, err, "failed to marshal json for rolebinding")
+		assert.Equal(t, expectedRoleBinding(), string(rbJson))
```

Comment:
I guess that the logic in these two verifications can be used as a custom assertion and call it only for the expected string/type